blockchain/contracts/StakingNFT.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/interfaces/IERC721.sol";
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract StakingContract is IERC721Receiver {
+    IERC721 public nft;
+
+    struct Stake {
+        uint256 tokenId;
+        uint256 amount;
+        uint256 timestamp;
+    }
+
+    event NFTStaked(address owner, uint256 tokenId, uint256 value);
+    event NFTUnstaked(address owner, uint256 tokenId, uint256 value);
+    event Claimed(address owner, uint256 amount);
+
+    // map staker address to stake details
+    mapping(address => Stake) public stakes;
+
+    // map staker total staking time
+    mapping(address => uint256) public stakingTime;
```

Comment:
Why are we storing this time?

Reply:
figured it could be useful for an emergencyWithdraw type of situation
eg assuming we need a contract interaction to allocate rewards to be claimed by the user as well as allowing to unstake after the game is finished, if there's an issue on that regard, user could be allowed to unstake after a certain period of time relative to stakingTime, even if the contract interaction that was meant to occur didn't happen